internal/gengapic/custom_operation.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
+// customOp represents a custom operation type for long running operations.
+type customOp struct {
+	proto     *descriptor.DescriptorProto
+	generated bool
```

Comment:
My one nit in the entire PR are about this field name. It may be clearer what it means to use `message`.

Reply:
Good call! Done.